File: leetcode/java/repeatedCharacter.java
// https://leetcode.com/problems/first-letter-to-appear-twice/description/
// 2351. First Letter to Appear Twice

сlass Solution {
    public char repeatedCharacter(String s) {
        HashMap<Character,Integer> map = new HashMap<>();
        for (int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            if (map.containsKey(c)) {
                 return c;
            }
        }
        return '\0';
    }
}


class RepeatedCharacter {
    public char repeatedCharacter(String s) {
        Set<Character> map = new HashSet<>();

        for (char c : s.toCharArray()) {
            if (map.contains(c)) {
                return c;
            }
            map.add(c);
        }

        return 'd';
    }
}

File: leetcode/repeatedCharacter.ts
//https://leetcode.com/problems/first-letter-to-appear-twice/description/

function repeatedCharacter(s: string): string {
    const map = {}
    for (let i = 0, len = s.length; i < len; i++) {
        const curr = s[i];
        map[curr] = (map[curr] || 0) + 1
        if (map[curr] == 2) return curr
    }
};


function repeatedCharacter2(s: string): string {
    const charSet = new Set()
    for (let i = 0, len = s.length; i < len; i++) {
       if (charSet.has(s[i])) {
            return s[i]
       }
       charSet.add(s[i])
    }
};
